coral/src/app/features/topics/details/subscriptions/components/TopicSubscriptionsDetailsModal.tsx

```diff
@@ -0,0 +1,137 @@
+import { useQuery } from "@tanstack/react-query";
+import { Modal } from "src/app/components/Modal";
+import { useAuthContext } from "src/app/context-provider/AuthProvider";
+import {
+  getAivenServiceAccountDetails,
+  getConsumerOffsets,
+} from "src/domain/acl/acl-api";
+import {
+  ConsumerOffsets,
+  ServiceAccountDetails,
+} from "src/domain/acl/acl-types";
+import { AclOverviewInfo } from "src/domain/topic/topic-types";
+
+interface TopicSubscriptionsDetailsModalProps {
+  closeDetailsModal: () => void;
+  isAivenCluster: boolean;
+  selectedSub: AclOverviewInfo;
+  offsetsData?: ConsumerOffsets;
+  serviceAccountData?: ServiceAccountDetails;
+}
+
+const TopicSubscriptionsDetailsModal = ({
+  closeDetailsModal,
+  isAivenCluster,
+  selectedSub,
+}: TopicSubscriptionsDetailsModalProps) => {
+  const user = useAuthContext();
+  const { environment, consumergroup, topicname, req_no } = selectedSub;
+
+  const {
+    data: offsetsData,
+    // @TODO: handle loading / error state gracefully
+    // isError: offsetsIsError,
+    // error: offsetsError,
+    // isFetched: offsetDataFetched,
+  } = useQuery(["consumer-offsets", environment, consumergroup], {
+    queryFn: () => {
+      if (environment !== undefined && consumergroup !== undefined) {
+        return getConsumerOffsets({
+          topicName: topicname,
+          env: environment,
+          consumerGroupId: consumergroup,
+        });
+      }
+    },
+    enabled:
+      selectedSub.environment !== undefined &&
+      selectedSub.consumergroup !== undefined,
+  });
+
+  const {
+    data: serviceAccountData,
+    // @TODO: handle loading / error state gracefully
+    // isError: serviceAccountIsError,
+    // error: serviceAccountError,
+    // isFetched: serviceAccountDataFetched,
+  } = useQuery(
+    [
+      "getAivenServiceAccountDetails",
+      environment,
+      topicname,
+      user?.username,
+      req_no,
+    ],
+    {
+      queryFn: () => {
+        if (
+          environment !== undefined &&
```

Comment:
Do we need this additional `if`? Only setting `enabled` to true for the same condition should do the same thing I think?

Reply:
It is not! This data fetching will look very different in the follow up PR to add the UI for the modal, though :o So I'm not sure it's worth addressing. This is a link to the Modal from the upcoming PR: https://github.com/aiven/klaw/blob/17a9cf34d7f280ba1129d8f4b99376aa3ed1c51b/coral/src/app/features/topics/details/subscriptions/components/TopicSubscriptionsDetailsModal.tsx